### ISSUE TITLE
Modify entity through PATCH from UI controls.

### ISSUE THREAD

**Issue by rahul7720** [object/Simple.OData.Client#386]
Hi,

I'm using Simple Odata Client to perform CRUD operations in a WPF application from WebApi Odata V4 Service.
I have a parent & a child entity:
```
public class Order
{
    public int OrderId{get;set;}
    public int Description{get;set;}
    public ObservableCollection<OrderLine> OrderLines {get;set;}
}
public class OrderLine
{
    public int OrderId{get;set;}
    public int OrderLineId{get;set;}
    public int ItenId{get;set;}
    public int ItemDescription{get;set;}
    public virtual Order Order {get;set;}
}
```

```
public class ManageOrders
{
  //Implements INotifyPropertyChanged
  public Order Order;

  public void GetOrder()
  {
      this.Order = await client
              .For<Order>()
              .ByKey(1001)
              .Expand(x.OrderLines)
              .FindEntriesAsync();
  }

  public void Save()
  {
      //If it's a new record
      if ("NEW")
      {
          // Add new item and save

          //Save Order
          this.Order = await client
                  .For<Order>()
                  .Set(this.Order)
                  .InsertEntryAsync();

          //Save Order Lines
          this.Order.OrderLines = await client
                          .For<OrderLine>()
                          .Set(this.Order.Lines)
                          .InsertEntryAsync();
      }
      //if it's existing record
      if ("MODIFIED")
      {
          this.Order = await client
                  .For<Order>()
                  .Set(this.Order)
                  .UpdateEntriesAsync();

          //Save Order Lines
          this.Order.OrderLines = await client
                          .For<OrderLine>()
                          .Set(this.Order.Lines)
                          .UpdateEntriesAsync();
      }
  }

  public void Delete()
  {
      //Delete
  }

}
```
I'm binding the Order properties to text box in a WPF window, and Order.OrderLines to DataGrid.

When I click the GET Button, Order will be fetched from DB. Then I change the data in Order and OrderLines. Then I Delete a OrderLine and add two new OrderLines.

As I'm using ObservarbleCollection, the changes will be automatically added to Source from UI.

**Requirement**

When I click SAVE button all the changes should be submitted to the server. (Batch request is preferred).

**Question/Issue**

How can I send only changed entities to the server through a PATCH request, without sending unmodified properties both in header and lines?

In other words, I have to do all the changes locally, then i have to submit it to server.
